operator/pkg/controller/karmada/controller.go

```diff
@@ -102,38 +92,18 @@ func (ctrl *Controller) Reconcile(ctx context.Context, req controllerruntime.Req
 		return ctrl.removeFinalizer(ctx, karmada)
 	}
 
+	if err := ctrl.validateKarmada(ctx, karmada); err != nil {
+		klog.Errorf("Validation failed for karmada: %+v", err)
+		return controllerruntime.Result{}, nil
+	}
```

Comment:
@jabellard 
There is a bug if we validate the object before handling the deletion events.
Let's say you have already run a Karmada instance with a configuration of `Karmada`, then you update the configuration which raises a validation error. In this situation, you can't remove the `Karmada` object before fixing the validation error, as each reconciliation will fail due to the validation, no chance to remove the finalizer.

Don't get me wrong, it's not blame just want you to know, in case it already affects you in your environments.